.github/workflows/gh-ci.yaml

```diff
@@ -26,51 +26,37 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, ]
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
         include:
           - os: windows-latest
             python-version: "3.12"
           - os: macos-latest
             python-version: "3.12"
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
 
-    - name: setup_miniconda
-      uses: conda-incubator/setup-miniconda@v2
+    - uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-        auto-update-conda: true
-        channel-priority: flexible
-        channels: conda-forge
-        add-pip-as-python-dependency: true
-        use-mamba: true
-        miniforge-variant: Mambaforge
 
-    - name: install package deps
+    - name: install optional deps
       run: |
-        mamba install pytest pytest-xdist pytest-cov codecov pbr numpy tqdm
-
-    - name: check install
-      run: |
-        which python
-        which pip
-        conda info
-        conda list
+        pip install pytest pytest-xdist pytest-cov codecov
```

Comment:
Very minor - if you using CodeCov's official-ish action, I don't think you need to install their Python package

Reply:
thanks!